crates/aptos-build-info/src/lib.rs

```diff
@@ -85,20 +95,20 @@ pub fn get_build_information() -> BTreeMap<String, String> {
     // Get Git metadata from environment variables set during build-time.
     // This is applicable for docker based builds  where the cargo cannot
     // access the .git directory, or to override shadow_rs provided info.
-    if let Ok(git_sha) = std::env::var("GIT_SHA") {
-        build_information.insert(BUILD_COMMIT_HASH.into(), git_sha);
+    if let Some(git_sha) = std::option_env!("GIT_SHA") {
+        build_information.insert(BUILD_COMMIT_HASH.into(), git_sha.into());
```

Comment:
I think it's the comment that's misleading. It's read during runtime within the Docker container.

https://github.com/aptos-labs/aptos-core/blob/b55ad61f7028b1875719b7c706b378e3b34b85cd/docker/builder/validator-testing.Dockerfile#L57-L65

Reply:
That doesnt make sense for build information tho...? Shouldn't this all be set at compile time?

Why would it make any sense for this to be overridden at runtime?

Reply:
I think my thought at that time was it's more flexible to set these env variables inside the output image, so we can use it elsewhere outside of aptos-node binary if need be. 

we can bake them in at compile time as well, but we need to make sure these variables are exposed in the builder step for your change to work.